tf2/CMakeLists.txt

```diff
@@ -6,6 +6,10 @@ if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 14)
 endif()
 
+if(NOT WIN32)
```

Comment:
I don't think we want to do this check this way for two reasons:

1.  This is not how we do it in most other packages.  Most other packages use: `if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")`.
1.  If a compiler other than gcc or clang is used on Linux (like some kind of cross-compiler), it will pick these flags up, potentially inappropriately.

Thus, I'll suggest we use the positive test, i.e. `if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")`.

Reply:
I agree, I copy pasted the logic from the wrong place :joy:.

Reply:
Done in https://github.com/ros2/geometry2/pull/322/commits/606b511024633a6ceca341fbe99598fdc93cf5e5.